app/controllers/concerns/application_shared.rb

```diff
@@ -21,4 +21,66 @@ def set_timezone
   def current_permission
     [action_permission, controller_permission].join('_')
   end
+
+  def set_taxonomy
+    user = User.current
+    return if user.nil?
+
+    ['location', 'organization'].each do |taxonomy|
+      return unless Taxonomy.enabled?(taxonomy.to_sym)
+
+      available = user.send("my_#{taxonomy.pluralize}")
+      determined_taxonomy = nil
+
+      # this keeps "any context" for admins unless they ask for specific context via parameters
+      if api_request? && user.admin?
```

Comment:
After this change, as an admin, you can't set `any context` in case you have set a default taxonomy. I recommend extracting this method to two: for the ui and for the api behavior, as I believe it would make it more obvious that there is such an issue

Reply:
I suppose by default you mean hammer defaults, not user's default organization. For admins, that should be ignored in this version completely. Org admin always has any context and can set specific context by using --organization-id and --location-id params.

Any context would normally be accessible by `--organization-id nil` but hammer currently can't set nil value. I'm not sure if extracting to a separate method would help, I wanted to keep all the logic on one place since it's "if condition hell" already. Jumping between methods does not add much readability.

EDIT: sry I was confused by the line you added the comment to, I was able to reproduce this in UI, the issue is that we fallback to default taxonomy with every request, while we should do this only on login for UI, defaulting makes sense for API requests